qualang_tools/macros/long_wait.py

```diff
@@ -0,0 +1,46 @@
+from typing import Union
+
+import numpy as np
+from qm.qua import declare, wait, for_
+from qualang_tools.addons.calibration.calibrations import u
+
+MAX_WAIT = 2**31 - 1  # units of clock cycles (4ns)
+
+
+def long_wait(wait_time: Union[float, int], *elements: str, threshold_for_looping: int = u.to_clock_cycles(1e6)):
+    """Perform a QUA wait command for longer than the current maximum wait time, 2^31-1.
+
+    This macro unrolls into QUA for-loop of smaller waits if the maximum wait time is exceeded in `wait_time`.
+
+    Note: `wait_time` must be a python number.
+
+    Args:
+        wait_time (Union[float, int]): Desired wait time in units of clock cycles (4ns).
+        *elements (str): Variable length argument list of elements to wait on.
+        threshold_for_looping: Minimum wait time at which `long_wait` uses a QUA for-loop of waits (default 1ms)
```

Comment:
I would add a higher bound here, because the whole point of this function is that there is a maximum wait time

Reply:
So the reason I did this is because constants above 1ms was causing long compile times. I wrote a unittest for a 30s `long_wait` using this value as the threshold and it worked well. When I was using `MAX_WAIT` as the threshold, it took really long to compile. Have you experienced this before?

Reply:
I see, yeah it already happened to me but I never really understood why...
I am also using 1ms as a safe threshold so it is fine :)

Reply:
You make a good point though. Let me add it as a comment to explain why it's 1ms and not higher